### PR TITLE
chore: add missing array types for 3.1

### PR DESCRIPTION
**Pull request by AntonKozachuk**
## What/Why/How?
Add missing Array props to type tree: https://json-schema.org/understanding-json-schema/reference/array.html

## Testing
N/a

## Screenshots (optional)
N/a

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/types/oas3_1.ts [hunk 1]
```diff
@@ -127,6 +127,8 @@ const Schema: NodeType = {
     dependentSchemas: listOf('Schema'),
     prefixItems: listOf('Schema'),
     contains: 'Schema',
+    minContains: { type: 'number' },
```
Comment (adamaltman) [inline packages/core/src/types/oas3_1.ts:130]: Is there an integer type?

Comment (AntonKozachuk) [reply packages/core/src/types/oas3_1.ts:130]: @adamaltman looks like we used it as type in 2.0, but don't have it starting from 3.0. Maybe @RomanHotsiy has more context regarding this.

Comment (RomanHotsiy) [reply packages/core/src/types/oas3_1.ts:130]: We should use `integer` and fix 3.0 schema usage too.

Comment (AntonKozachuk) [reply packages/core/src/types/oas3_1.ts:130]: @RomanHotsiy I rechecked and looks like the integer type in 2.0 was used according to the spec. And in 3.0 and older there seem no fields with type integer, besides the `type` itself.

Example:
image

Comment (RomanHotsiy) [reply packages/core/src/types/oas3_1.ts]: OAS3 and later Schema type is covered by JsonSchema. See here: https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.3.1

**Review by RomanHotsiy — APPROVED** [review]
LGTM 🚢

**Review by tatomyr — APPROVED** [review]

**Pull request merged**